### PR TITLE
Align article meta and body image

### PR DESCRIPTION
**Pull request by ruicosta042**
## What does this change?

Screenshot 2019-12-23 at 13 12 41

Features:
- 100% CSS flexbox layout of article structure
- flat structure of article's elements (less markup)
- no conditional rendering of elements to simulate reordering (less code duplication)
- compatibility with >IE10

Fixes:
- always aligns ArticleMeta with MainMedia element
- always aligns article's ads column with Standfirst (showcase layout)
- no client JS to position the ads column on the right side
- Standfirst always below MainMedia on IE (showcase layout)

## Link to supporting Trello card
https://trello.com/c/1P7ezW4F/847-align-article-meta-and-body-image

### CODE REVIEW

**Comment by shtukas** [comment]
+1 in principle (and feel free to merge if it's urgent), but would be nice to hear from @oliverlloyd first :)

File: src/web/components/GuardianLines.stories.tsx [hunk 1]
```diff
@@ -58,26 +54,33 @@ export const eightLines = () => {
...
+                        pillar="news"
+                    />
+                </ArticleContainer>
             </Section>
         </>
     );
 };
 eightLines.story = { name: 'with eight lines' };
 
+// @ts-ignore
```
Comment (oliverlloyd) [inline src/web/components/GuardianLines.stories.tsx]: Ditto on ts-ignore. Stories should be test cases for how the component works so if typescript is failing that needs to be fixed.

File: src/web/components/RightColumn.tsx [hunk 2]
```diff
@@ -1,35 +1,53 @@
...
     }
 `;
 
-const padding = css`
-    padding-top: 6px;
-    /* Use margin here to the prevent flicker on load that you get
-       with padding (because the header section has flex-grow: 1) */
-    margin-left: 10px;
+const rightColumnStyles = css`
+    position: absolute; /* flow out of container */
```
Comment (oliverlloyd) [inline src/web/components/RightColumn.tsx]: It _could be_ this and/or the `height: 0px` above that is causing the Cypress tests to fail. The `MostViewedRight` componmenmt is not being rendered anymore.

File: src/web/layouts/Layouts.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+};
+/* tslint:enable */
+
+export const Standard = () => (
+    <StandardLayout CAPI={CAPI as CAPIType} NAV={NAV as NavType} />
+);
+Standard.story = { name: 'Standard' };
+
+export const Showcase = () => (
+    // @ts-ignore
```
Comment (oliverlloyd) [inline src/web/layouts/Layouts.stories.tsx]: ditto `ts-ignore`

File: src/web/layouts/Layouts.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+import { StandardLayout } from './StandardLayout';
+import { ShowcaseLayout } from './ShowcaseLayout';
+
+/* tslint:disable */
+export default {
+    component: StandardLayout,
+    title: 'Layouts/Standard',
+    parameters: {
+        percy: { widths: [600, 979, 1139, 1299, 1400] },
```
Comment (oliverlloyd) [inline src/web/layouts/Layouts.stories.tsx]: percy is no longer used

File: src/web/components/ArticleStandfirst.stories.tsx [hunk 5]
```diff
@@ -17,17 +17,30 @@ export default {
...
-                    <ArticleStandfirst
-                        standfirst="This the default standfirst text. Aut explicabo officia delectus omnis repellendus voluptas"
-                        pillar="news"
-                    />
-                </ArticleContainer>
-            </Flex>
+            <ArticleContainer>
+                <ArticleTitle
+                    CAPI={
+                        // @ts-ignore
```
[the remark this answers is not in the document]
Comment (oliverlloyd) [reply src/web/components/ArticleStandfirst.stories.tsx]: I've actually already done this refactor [here](https://github.com/guardian/dotcom-rendering/pull/1038/commits/6fad44cd216c02532cc14db2f903859237894e63)

**Comment by ruicosta042** [comment]
> This PR builds on top of that existing pattern.

As stated on the comment, this PR builds on top of the existing structure. Then, it fixes the issue identified on the ticket. If further issues are found related to component architecture, those should be solved on its own new ticket.

**Comment by oliverlloyd** [comment]
I created [a branch](https://github.com/guardian/dotcom-rendering/tree/oliver/grid-layout) that demonstrates the concept of using wrappers to separate the layout logic from the article components. (Note. It's just a demo branch, it has bugs)

```typescript
                <ShowcaseGrid>
                    <GridItem area="title">
                        <ArticleTitle
                            CAPI={CAPI}
                            badge={GE2019Badge}
                            inLeftCol={true}
                        />
                    </GridItem>
                    //  ...etc
```

As you can see, the layout logic is contained in the wrapper `GridItem` this means there are no code changes to each child component.

I also extracted `isShowcase` by using a similar wrapper pattern. Before, `ArticleHeadline` shifted itself up or down depending on if it was showcase or not. Now that layout logic has been lifted up away from the headline component.

```typescript
                    <GridItem area="headline">
                        <Shift direction="down" by="100px">
                            <ArticleHeadline
                                headlineString={CAPI.headline}
                                designType={CAPI.designType}
                                pillar={CAPI.pillar}
                                webPublicationDate={CAPI.webPublicationDate}
                                tags={CAPI.tags}
                                byline={CAPI.author.byline}
                            />
                        </Shift>
                    </GridItem>
```

Having all the layout logic in one place makes it easier to reason about what's going on. In this particular branch, I went with css grid to do the layout so we had a comparison but the extraction principle remains the same - you could port this  branch to use flex wrapping fairly easily. Personally, I find grid a nice readable solution here that doesn't depend on fixed widths but this isn't a flex vs. grid discussion.

**Pull request closed without merging**